torchchat/generate.py

```diff
@@ -958,16 +959,19 @@ def chat(
             if get_system_prompt == "y" or get_system_prompt == "Y":
                 self.system_prompt = input("What is your system prompt? \n")
 
-        # elif not generator_args.is_torchtune_model:
-        #     max_seq_length = min(
-        #         encoded.size(0) + generator_args.max_new_tokens,
-        #         (
-        #             text_transformer_args.block_size
-        #             if text_transformer_args is not None
-        #             else 2048
-        #         ),
-        #         max_seq_length,
-        #     )
+        # `is_torchtune_model` is a misnomer since it doesn't capture all
+        # torchtune models (i.e. Flamingo)
+        # See Issue: https://github.com/pytorch/torchchat/issues/1273
+        elif not generator_args.is_torchtune_model and self.model.config.model_type != ModelType.Flamingo:
+            max_seq_length = min(
+                encoded.size(0) + generator_args.max_new_tokens,
+                (
+                    text_transformer_args.block_size
+                    if text_transformer_args is not None
+                    else 2048
```

Comment:
discrete numbers and strings should be constants or at the very least have comments explaining why that number.